common/lib/xmodule/xmodule/tabs.py

```diff
@@ -172,6 +172,10 @@ def __ne__(self, other):
         """
         return not self == other
 
+    def __hash__(self):
```

Comment:
in py3 __hash__ call fails with error. so we have to explicitly define those.

Reply:
How did you arrive at using `repr` as the base for the hash.  Does the repr contain all the unique info for a tab?  Can you providde an example of what the `repr` looks like?

Reply:
@feanil it represents '<lms.djangoapps.discussion.plugins.DiscussionTab object at 0x7f9af9f7e518>'. As it includes memory location which will be different for all objects. So hash will be unique.

Reply:
Right but if the hash is stored anywhere, it will change each time the process is instantiated which is not what we want because that will make comparing to see if two tabs are the same impossible.  We want two instances of the same tab to hash to the same thing.

@dianakhuang @bmedx  I think you guys were looking at tab related code recently, can you take a look at this and suggest a good approach?

Reply:
@feanil still waiting.

Reply:
Very sorry for the delay. So I believe this is safe, the values used to hash in 2.7 were also transient and somewhat based on memory location so this shouldn't be introducing anything new.